### PR TITLE
React: Cache previously selected visible columns 

### PR DESCRIPTION
**Pull request by jennyziyi-xu**
- This PR only addresses issues within a gene search. If a second gene search is performed, the default visible columns is set to be all the columns in the group "Variant". However, if we want to conserve visible columns between gene searches, this issue will be addressed in another PR.  

- In the result table, collapsing a group will make all columns in the group invisible. Expanding a group will show only previously visible groups. 
- In the "Column VIsibility" pop-up, unchecking a group will uncheck all the columns in the group. Checking a group will only show cached visible columns ( in the case that there are no cached visible columns, all the columns in the group will be shown). Closing the "Column Visibility" pop-up will not save the changes made in the pop-up. 
- In summary, there is a cache `cacheTable` for the result table and a cache `cacheVisibility` for the "Column Visibility" pop-up. Every time user opens the "Column Visibility" modal, `cacheVisibility` is initialized to be equal to `cacheTable`. Every time user clicks `apply` in the modal, `cacheTable` is updated to be equal to `cacheVisibility`.

### CODE REVIEW

File: react/src/components/Table/ColumnVisibilityModal.tsx [hunk 1]
```diff
@@ -1,34 +1,74 @@
...
+        return !checkedColumns[groupMapping[header]];
+    };
+
+    // Find a column's group: "Variant", "Variant Details" or "Case Details".
+    const columnToGroup = (columnId: string) => {
+        let group = headerGroups[0].headers[0];
+        headerGroups[0].headers.forEach(header => {
+            if (header.columns?.find(c => c.id === columnId)) group = header;
+        });
+        return group;
```
Comment (kevinlul) [inline react/src/components/Table/ColumnVisibilityModal.tsx]: So this will default to the first column?

Comment (jennyziyi-xu) [reply react/src/components/Table/ColumnVisibilityModal.tsx]: Yes, since each column belongs to a group, the default group should not cause any issues. 

It's better to remove this default group though, and I just removed it.

**Review by themisteryang — APPROVED** [review]
<details>
  <summary>LGTM, the bug looks to be fully fixed!</summary>
  Just for reference, the way to reproduce the bug was:
  
  1. Start (by default) with all the columns checked.
  2. Uncheck `Chr` and `Start`.
  3. Uncheck `Variant` (this unchecks all of the `Variant` columns, as expected).
  4. Check `Chr` (this also checks `Variant`, as expected).
  5. Uncheck `Variant` again (this unchecks all of the `Variant` columns, as expected).
  6. Check `Variant`.
  
  Now, every `Variant` column except for `Start` is checked instead of only `Chr` being checked.
</details>
